### PR TITLE
feat: Document the User Feedback API

### PR DESCRIPTION
**Pull request by philipphofmann**
Add more context on how to implement the user feedback API on user facing platforms.

Fixes GH-218

### CODE REVIEW

File: src/docs/sdk/features.mdx [hunk 1]
```diff
@@ -75,9 +75,22 @@ Ability to get the ID of the last event sent. Event IDs are useful for correlati
...
+[Java](https://docs.sentry.io/platforms/java/enriching-events/user-feedback/).
+
+On mobile and desktop, it is common to prompt the user for feedback after a crash happened on the previous run of the application. Therefore the SDKs should
+implement the `onCrashedLastRun` callback on the options. This callback gets called shortly after the initialization of the SDK when the last program execution
+terminated with a crash. The SDK should execute the callback only once during the entire run of the program to avoid multiple callbacks if there are multiple 
+crash events to send.
+
+### Backend Platforms
+
+On backend platforms, SDKs should document how to use the last event ID to prompt the user for feedback themselves.
```
Comment (bruno-garcia) [inline src/docs/sdk/features.mdx:93]: What about we link to some code:

```suggestion
### Backend Platforms

On backend platforms, SDKs should document how to use the last event ID to prompt the user for feedback themselves. 

### Example implementations

User Feedback class:
 * [Objective-C](https://github.com/getsentry/sentry-cocoa/blob/9eedc425727f0daccca1bd2be6021f6c3d9c654c/Sources/Sentry/SentryUserFeedback.m)
 * [Java](https://github.com/getsentry/sentry-java/blob/671f9e0b8b709ef18e8b0788c31df44f99d09d35/sentry/src/main/java/io/sentry/UserFeedback.java#L7)
 * [C#](https://github.com/getsentry/sentry-dotnet/blob/6ddf8363170bffee74a8c3cbf44335797d39ca9c/src/Sentry/Protocol/UserFeedback.cs#L10)

Envelope item:

 * [Objective-C](https://github.com/getsentry/sentry-cocoa/blob/fd3e46efe59324e894d5601a7b3028824a588dda/Sources/Sentry/SentryEnvelope.m#L145-L162)
 * [Java](https://github.com/getsentry/sentry-java/blob/b2e8dd43ea2ef37201b215ec409655e49e3231c8/sentry/src/main/java/io/sentry/SentryEnvelopeItem.java#L131-L155)
 * [C#](https://github.com/getsentry/sentry-dotnet/blob/3bc6c66a14e653982a6a25df033d68d8ad12c4af/src/Sentry/Protocol/Envelopes/EnvelopeItem.cs#L170-L178)
 
```

File: src/docs/sdk/features.mdx [hunk 2]
```diff
@@ -75,9 +75,36 @@ Ability to get the ID of the last event sent. Event IDs are useful for correlati
 
 ## User Feedback
 
-On user-facing platforms such as mobile or the browser this means first class support for requesting user feedback when an error occurs.
+For all SDKs, it is strongly recommended to send the user feedback as an [envelope](/sdk/envelopes/#user-feedback). Alternatively, the SDKs can
+use the [user feedback endpoint](https://docs.sentry.io/api/projects/submit-user-feedback/), which is not recommended.
```
Comment (marandaneto) [inline src/docs/sdk/features.mdx:79]:
```suggestion
For all SDKs, it is strongly recommended to send the `User Feedback` as an [envelope](/sdk/envelopes/#user-feedback). Alternatively, the SDKs can
use the [User Feedback endpoint](https://docs.sentry.io/api/projects/submit-user-feedback/), which is not recommended.
```

File: src/docs/sdk/features.mdx [hunk 3]
```diff
@@ -75,9 +75,36 @@ Ability to get the ID of the last event sent. Event IDs are useful for correlati
...
 
-On backend platforms, SDKs should document how to use the last event ID to prompt the user for feedback themselves.
+### User Facing Platforms
+On user-facing platforms such as mobile, desktop, or browser this means first-class support for requesting user feedback when an error or crash occurs.
```
Comment (marandaneto) [inline src/docs/sdk/features.mdx:82]:
```suggestion
On user-facing platforms such as mobile, desktop, or browser this means first-class support for requesting User Feedback when an error or crash occurs.
```

File: src/docs/sdk/features.mdx [hunk 4]
```diff
@@ -75,9 +75,36 @@ Ability to get the ID of the last event sent. Event IDs are useful for correlati
...
+[Java](https://docs.sentry.io/platforms/java/enriching-events/user-feedback/).
+
+On mobile and desktop, it is common to prompt the user for feedback after a crash happened on the previous run of the application. Therefore the SDKs should
+implement the `onCrashedLastRun` callback on the options. This callback gets called shortly after the initialization of the SDK when the last program execution
+terminated with a crash. The SDK should execute the callback only once during the entire run of the program to avoid multiple callbacks if there are multiple 
+crash events to send.
+
+### Backend Platforms
+
+On backend platforms, SDKs should document how to use the last event ID to prompt the user for feedback themselves.
```
Comment (marandaneto) [inline src/docs/sdk/features.mdx:93]: L: maybe we link the `last event ID` to the docs? I guess it might be described in the unified API docs

**Review by marandaneto — APPROVED** [review]

**Pull request merged**